src/graph.js

```diff
@@ -40,35 +40,100 @@ class Graph {
   // Optionally accepts an array of other GraphNodes for the new vertex to be connected to
   // Returns the newly-added vertex
   addVertex(value, edges = []) {
+    const vertex = new GraphNode({ value, edges });
 
+    this.vertices.push(vertex);
+    if (this.vertices.length === 2) {
+      this.addEdge(this.vertices[0], vertex);
+    }
+    if (edges.length > 0) {
+      edges.forEach((edge) => {
+        this.addEdge(vertex, edge);
+      });
+    }
+
+    return vertex;
   }
   // Checks all the vertices of the graph for the target value
   // Returns true or false
   contains(value) {
+    for (let i = 0; i < this.vertices.length; i++) {
+      if (this.vertices[i].value === value) return true;
+    }
+
+    return false;
+  }
 
+  getVertexIndex(vertex) {
```

Comment:
I'm not sure this will work if you pass in a reference to a GraphNode instance, but I may be wrong about that.